src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx

```diff
@@ -163,6 +150,110 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
+      moduleLogger.error({ fn: 'getSendInput', e }, 'Error building THORChain savers Tx')
+    }
+  }, [
+    accountId,
+    assetId,
+    getEstimateFeesArgs,
+    state?.deposit.cryptoAmount,
+    state?.deposit?.estimatedGasCrypto,
+    asset,
+    maybeAccountAddress,
+    selectedCurrency,
+  ])
+
+  const handleMultiTxSend = useCallback(async (): Promise<string | undefined> => {
+    if (!walletState.wallet) return
+
+    // THORChain Txs need to always be sent from the same address, since the address (NOT the pubkey) is used to identify an active position
+    // The way THORChain does this is by not being xpub-compliant, and only exposing a single address for UTXOs in their UI
+    // All deposit/withdraws done from their UI are always done with one/many UTXOs from the same address, and change sent back to the same address
+    // We also do this EXCLUSIVELY for THORChain Txs. The rest of the app uses xpubs, so the initially deposited from address isn't guaranteed to be populated
+    // if users send other UTXO Txs in the meantime after depositing
+    // Additionally, we select their highest balance UTXO address as a first deposit, which isn't guaranteed to contain enough value
+    //
+    // For both re/deposit flows, we will possibly need a pre-Tx to populate their highest UTXO/previously deposited from address with enough value
+
+    const depositInput = await getDepositInput()
+    if (!depositInput) throw new Error('Error building send input')
+
+    let txId: string
+
+    try {
+      // 1. Try to deposit from the originally deposited from / highest UTXO balance address
+      // If this is enough, no other Tx is needed
+      txId = await handleSend({
+        sendInput: depositInput,
+        wallet: walletState.wallet,
+      })
+    } catch (e) {
+      // 2. signAndBroadcastTransaction threw, meaning there's not enough value in the picked address - send funds to it
+      const preDepositInput = await getPreDepositInput()
+      if (!preDepositInput) throw new Error('Error building send input')
+      txId = await handleSend({
+        sendInput: preDepositInput,
+        wallet: walletState.wallet,
+      })
+      // 3. Sign and broadcast the depooosit Tx again
+      txId = await handleSend({
+        sendInput: depositInput,
+        wallet: walletState.wallet,
+      }).catch(_e => '')
+    }
+
+    return txId
+  }, [getDepositInput, getPreDepositInput, walletState.wallet])
+  useEffect(() => {
+    if (!accountId) return
+    ;(async () => {
+      const accountAddress = isUtxoChainId(chainId)
+        ? await getThorchainSaversPosition(accountId, assetId)
+            .then(({ asset_address }) =>
+              chainId === bchChainId ? `bitcoincash:${asset_address}` : asset_address,
+            )
+            .catch(async () => {
+              const addressesWithBalances = await getAccountAddressesWithBalances(accountId)
+              const highestBalanceAccount = addressesWithBalances.sort((a, b) =>
```

Comment:
Since we have to arbitrarily pick an UTXO address for the initial deposit that will be reused, might as well pick the highest balance, so there's at least a change it has enough value

Reply:
ackkshually, this is a case where we may want to make this a hook. `useMaybeFromUTXOAccountAddress` as it's used in two places?

Reply:
derp rebase issue this is the same file lol

Reply:
Today on "I just got rugged by rebasing" 0e1fa3eeb4e69400e83614c4bcd9c20e912ff73c

Reply:
we can simplify to avoid passing in `chainId` too